### ISSUE TITLE
8$: Warn on unused arrow expression

### ISSUE THREAD

**Issue by strager** [quick-lint/quick-lint-js#659]
```js
useEffect(() => {
  // BUG: async arrow function not used.
  async () => {
    const userReviewsForEachAudiobook = await Promise.all(
      reviewsUrlList.map((review) => {
        return fetch(review).then((res) => res.json());
      })
    );
  };
}, [reviewsUrlList]);
```

**Comment by singalhimanshu** [comment]
I claim this for-hire task. I expect payment after I complete this task. I will email the quick-lint-js team if I am assigned this task.

**Comment by singalhimanshu** [comment]
@strager which of these should report `error_unused_arrow_function`:
```js
// 1.
async => {}
async => rhs;
// 2.
let => {}
// 3.
x => {}
// 4.
=> x + y
```

**Comment by strager** [comment]
> @strager which of these should report `error_unused_arrow_function`:

1, 2, and 3.

4 should report a different diagnostic (E0105).

**Comment by jr-boney** [comment]
useEffect(() => {
  (async () => {
    const userReviewsForEachAudiobook = await Promise.all(
      reviewsUrlList.map((review) => {
        return fetch(review).then((res) => res.json());
      })
    );
  })();
}, [reviewsUrlList]);

**Comment by strager** [comment]
@jr-boney I don't see any unused arrow expressions in your code. Are you seeing a bug in quick-lint-js related to your code?
